src/components/course/EnrollButton.jsx

```diff
@@ -56,7 +57,8 @@ export default function EnrollButton() {
           course_id: key,
           enterprise_customer_uuid: enterpriseConfig.uuid,
           next: `${process.env.LMS_BASE_URL}/courses/${key}/course`,
-          failure_url: global.location,
+          // Redirect back to the LP dashboard with a failure query param
+          failure_url: `${global.location.host}/${enterpriseConfig.slug}/?${ENROLLMENT_FAILED}=true`,
```

Comment:
For some reason I was unable to trigger any failure in data sharing consent locally, so I'll test this more thoroughly on stage/prod, but I don't think this should be too hairy